addons/web/static/src/webclient/actions/action_service.js

```diff
@@ -692,11 +692,19 @@ function makeActionManager(env) {
                     } else {
                         const lastCt = controllerStack[controllerStack.length - 1];
                         if (lastCt) {
-                            // the error occurred while rendering a new controller,
-                            // so go back to the last non faulty controller
-                            // (the error will be shown anyway as the promise
-                            // has been rejected)
-                            restore(lastCt.jsId);
+                            if (lastCt.jsId === controller.jsId) {
+                                // the error occurred on the controller which is
+                                // already in the DOM, so simply show the error
+                                Promise.resolve().then(() => {
+                                    throw error;
+                                });
```

Comment:
Shouldn't we simply not call restore in this case since the controller we're trying to restore is already in the DOM? Seems like we're just replacing the current controller with a new instance of the same controller (otherwise this.isMounted would be true).

Reply:
side note:
```suggestion
                            if (lastCt.jsId === controller.jsId) {
                                // the error occurred on the controller which is
                                // already in the DOM, so simply show the error
                               Promise.reject(error);
```